RecipeApp.Core/RecipeConverter.cs

```diff
@@ -1,14 +1,176 @@
 ﻿using System;
 using System.Globalization;
+using System.Collections.Generic;
+using System.Linq;
 using System.IO;
 using System.Net;
 
 namespace RecipeApp
 {
     public class RecipeConverter
     {
-        static double[] multipliers = {240, 14.79, 4.93};
-        static string[] units = {"cup", "tablespoon", "teaspoon"};
+        static double[] multipliers = {240,3790,29.57,470,950,14.79,4.93};
+        static string[] units = {"cup","gallon","fluid ounce","pint","quart","tablespoon","teaspoon"};
+        static List<string> doubleUnits = new List<string>();
+
+        // Entry method to get some initial setup for Lists
+        public static void InitialSetup() {
+            PopulateDoubleUnitsList();
+            SortCookingUnitsByMultiplier();
+        }
+
+        public static string RecipeValidation(string recipeText) {
+            if (recipeText.Length == 0)
+            {
+                return "Empty recipes are not allowed.";
+            }
+            else if (!recipeText.Any(char.IsDigit))
+            {
+                return "Recipe does not contain any numeric values.";
+            }
+            else if (!RecipeConverter.DoesContainCookingUnits(recipeText))
+            {
+                return "Recipe does not contain any cooking units";
+            }
+            else if (RecipeConverter.ReturnFirstFaultyCookingUnitIndex(recipeText) != -1)
+            {
+                return "Bad or non-existant cooking unit values detected.";
+            }
+            return "";
+        }
+
+        // Sort cooking units by multiplier
+        static void SortCookingUnitsByMultiplier() {
+            bool sorted = true;
+            for (int j = 0; j < multipliers.Length; j++)
+            {
+                for (int i = 0; i < multipliers.Length - 1; i++)
+                {
+                    if (multipliers[i]>multipliers[i+1])
+                    {
+                        //swapping multipliers
+                        double temp = multipliers[i];
+                        multipliers[i] = multipliers[i + 1];
+                        multipliers[i + 1] = temp;
+                        //swapping units
+                        string strTemp = units[i];
+                        units[i] = units[i + 1];
+                        units[i + 1] = strTemp;
+
+                        sorted = false;
+                    }
+                }
+                if (sorted)
+                {
+                    return;
+                }
+            }
+        }
+
+        // Validation method to check if string contains any of the cooking units
```

Comment:
If you put trippled slash (`///`) it will create a method documentation with a summary comment. Prefer to use that, because it will allow reading the what the method does when you hover your mouse over it.